src/code/z_play_hireso.c

```diff
@@ -1,27 +1,1450 @@
+                                                    this->cursorPageRow + this->cursorPage, this->cursorEntry - 3)]);
+                    }
+                } else {
+                    if (play->msgCtx.currentTextId != sBombersNotebookTextIds[this->cursorPageRow + this->cursorPage]) {
+                        Message_ContinueTextbox(play, sBombersNotebookTextIds[this->cursorPageRow + this->cursorPage]);
+                    }
+                }
+            } else {
+                Message_CloseTextbox(play);
+            }
+
+            if (stickAdjY < -30) {
+                if (this->cursorPageRow < 3) {
+                    play_sound(NA_SE_SY_CURSOR);
+                    this->cursorEntry = 0;
+                    this->cursorPageRow++;
+                } else if (this->cursorPage < 16) {
+                    if (sStickYRepeatState == -1) {
+                        this->scrollAmount = -24;
+                    } else {
+                        this->scrollAmount = -12;
+                    }
+                    this->scrollOffset = 0;
+                    this->cursorEntry = 0;
+                }
+            } else if (stickAdjY > 30) {
+                if (this->cursorPageRow > 0) {
+                    play_sound(NA_SE_SY_CURSOR);
+                    this->cursorEntry = 0;
+                    this->cursorPageRow--;
+                } else if (this->cursorPage != 0) {
+                    play_sound(NA_SE_SY_CURSOR);
+                    this->cursorPage--;
+                    if (sStickYRepeatState == 1) {
+                        this->scrollAmount = 24;
+                    } else {
+                        this->scrollAmount = 12;
+                    }
+                    this->scrollOffset = -48;
+                    this->cursorEntry = 0;
+                }
+            }
+        } else if (this->scrollAmount < 0) {
+            this->scrollOffset += this->scrollAmount;
+            if (ABS_ALT(this->scrollOffset) >= 48) {
+                play_sound(NA_SE_SY_CURSOR);
+                this->scrollOffset = 0;
+                this->scrollAmount = 0;
+                this->cursorPage++;
+            }
+        } else {
+            this->scrollOffset += this->scrollAmount;
+            if (this->scrollOffset == 0) {
+                this->scrollOffset = 0;
+                this->scrollAmount = 0;
+            }
+        }
+    }
+}
+
+void BombersNotebook_Init(BombersNotebook* this) {
+    bzero(this, sizeof(BombersNotebook));
+}
+
+void BombersNotebook_Destroy(BombersNotebook* this) {
+    if (this->loadState == BOMBERS_NOTEBOOK_LOAD_STATE_STARTED) {
+        BombersNotebook_LoadFiles(this, OS_MESG_BLOCK);
+    }
+    if (this->scheduleDmaSegment != NULL) {
+        ZeldaArena_Free(this->scheduleDmaSegment);
+        this->scheduleDmaSegment = NULL;
+    }
+    //! @bug: Does not free malloc'd memory for schedule segment
```

Comment:
I wonder if opening and closing the bombers notebook many times could crash the game because of a memory leak

Reply:
Destroy actually doesn't get called when you just close the notebook. It gets called in play destroy. So you have to open the notebook, close, and than force play to reload.

Reply:
I don't think a memory leak is possible because Play_Init calls ZeldaArena_Init which I believe should free the whole thing for reuse.